#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *
import roslib.packages

gen = ParameterGenerator()

gen.add("roi_min_x", double_t, 0, "ROI: minimal x", 0.8, -5.0, 5.0)
gen.add("roi_max_x", double_t, 0, "ROI: maximal x", 0.95, -5.0, 5.0)
gen.add("roi_min_y", double_t, 0, "ROI: minimal y", -0.1, -5.0, 5.0)
gen.add("roi_max_y", double_t, 0, "ROI: maximal y", 0.1, -5.0, 5.0)
gen.add("roi_min_z", double_t, 0, "ROI: minimal z", 0.68, 0.0, 5.0)
gen.add("roi_max_z", double_t, 0, "ROI: maximal z", 0.90, 0.0, 5.0)

gen.add("detection_threshold", double_t, 0, "Threshold for detection", 0.93, 0.0, 1.0)
gen.add("learning_threshold", double_t, 0, "Threshold for learning", 0.90, 0.0, 1.0)
gen.add("minimum_object_distance", double_t, 0, "Minimal object distance in set of hypotheses", 4.0, 0.0, 10.0)
gen.add("base_link", str_t, 0, "base link", "/base_link")
gen.add("camera_link", str_t, 0, "camera link", "/camera_link")
gen.add("input_cloud_topic", str_t, 0, "input cloud topic", "/camera/depth_registered/points")
gen.add("database_file_name", str_t, 0, "database_file_name", "object_")
gen.add("database_folder_name", str_t, 0, "database_folder_name", str(roslib.packages.get_pkg_dir('mcr_object_recognition_linemod') + '/ros/data/'))
gen.add("publish_object_pointclouds", bool_t, 0, "Publish object point clouds when recognizing", True)
gen.add("mode_visualization", bool_t, 0, "Visualize objects found", True)
gen.add("num_modalities", int_t, 0, "Set the number of used modalities", 2, 1, 10)
gen.add("learnable_object_name", str_t, 0, "Object name to learn", "cappuccino")

exit(gen.generate("mcr_object_recognition_linemod", "mcr_object_recognition_linemod", "ObjectRecognitionLinemod"))
